video2dataset/data_reader.py

```diff
@@ -1,23 +1,47 @@
 """classes and functions for downloading videos"""
 import requests
 import tempfile
+import time
 import yt_dlp
 
 
-QUALITY = "360p"
 
-# TODO: update this to be link in video2numpy (with retries)
-def handle_youtube(youtube_url):
+def handle_youtube(youtube_url, retries, timeout, video_height, video_width):
```

Comment:
unrelated with this PR, but I think we should have a YtDlpDataReader
and a Mp4UrlDataReader

and then remove these handle_X functions

what do you think?

Reply:
we can do that or we can generalize it all to YtDlpDataReader since it's supposed to be able to handle everything. Then I guess we can make separate ones for the things it doesn't handle

Reply:
so yeah I think that's a good idea but I think Mp4UrlDataReader would work in YtDlpDataReader